homeassistant/components/rpi_power/binary_sensor.py

```diff
@@ -31,43 +34,22 @@ async def async_setup_entry(
 class RaspberryChargerBinarySensor(BinarySensorEntity):
     """Binary sensor representing the rpi power status."""
 
-    def __init__(self, under_voltage):
+    def __init__(self, under_voltage: UnderVoltage) -> None:
         """Initialize the binary sensor."""
         self._under_voltage = under_voltage
-        self._is_on = None
+        self._attr_is_on = None
+        self._attr_device_class = DEVICE_CLASS_PROBLEM
+        self._attr_icon = "mdi:raspberry-pi"
+        self._attr_name = "RPi Power status"
+        self._attr_unique_id = "rpi_power"  # only one sensor possible
         self._last_is_on = False
 
-    def update(self):
+    def update(self) -> None:
         """Update the state."""
-        self._is_on = self._under_voltage.get()
-        if self._is_on != self._last_is_on:
-            if self._is_on:
+        self.attr_is_on = self._under_voltage.get()
+        if self.attr_is_on != self._last_is_on:
```

Comment:
`self.attr_is_on` seems to be wrong, did you test this?

We don't need a global `self._last_is_on` for this, instead, fetch the value, compare it to the current value -> log the warning if changed -> store the state. That allows for removing `_last_is_on` completely from this class.

Reply:
I've edited files on my local pi. now everything should be fixed.